From eab57dae2b5260f8dea162a17415a4500b4bbb3c Mon Sep 17 00:00:00 2001
From: Yoshihito Ogawa <yoshihito.ogawa.kc@renesas.com>
Date: Wed, 5 Nov 2025 17:05:45 +0900
Subject: [PATCH] fix(driver): fix wayland driver

This commit fix a issue where mouse events cannot
be delivered to LVGL applications when the
applications start automatically with systemd.

Signed-off-by: Yoshihito Ogawa <yoshihito.ogawa.kc@renesas.com>
---
 src/drivers/wayland/lv_wayland.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/drivers/wayland/lv_wayland.c b/src/drivers/wayland/lv_wayland.c
index dc71f6677..b11d29453 100644
--- a/src/drivers/wayland/lv_wayland.c
+++ b/src/drivers/wayland/lv_wayland.c
@@ -2222,10 +2222,10 @@ skip:
 
 static void _lv_wayland_handle_input(void)
 {
-    int prepare_read = wl_display_prepare_read(application.display);
-    while(prepare_read != 0) {
+    while(wl_display_prepare_read(application.display) != 0) {
         wl_display_dispatch_pending(application.display);
     }
+    wl_display_flush(application.display);
 
     wl_display_read_events(application.display);
     wl_display_dispatch_pending(application.display);
-- 
2.34.1

